python/django_bolt/auth/tokens.py

```
@@ -0,0 +1,300 @@
+       revoked and the call fails.
+    3. If ``max_session_lifetime`` is set, ``now - oat`` must be within it.
+    4. On success: issue a new pair (Mode A / ``rotate=True``) and revoke the
+       old ``jti``, or issue only a new access token (Mode B /
+       ``rotate=False``) leaving the refresh token in place.
+
+    Raises ``TokenRotationError`` on any failure (with a generic message —
+    callers should return a uniform 401 to avoid an oracle).
+    """
+    jti = refresh_claims.get("jti")
+    if not jti:
+        raise TokenRotationError("Refresh token missing jti")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)